File: planik-android/app/src/main/java/pl/planik/local/entity/PlanEntity.kt
package pl.planik.local.entity

import androidx.room.*
import java.time.OffsetDateTime
import java.time.ZoneId

@Entity(
  tableName = "plans",
  foreignKeys = [
    ForeignKey(
      entity = UserEntity::class,
      parentColumns = ["id"],
      childColumns = ["user_id"],
      onDelete = ForeignKey.CASCADE,
      onUpdate = ForeignKey.CASCADE
    )
  ],
  indices = [
    Index(value = ["user_id"])
  ]
)
data class PlanEntity(
  @ColumnInfo(name = "id") @PrimaryKey(autoGenerate = true) val id: Int = -1,
  @ColumnInfo(name = "user_id") val userId: Int,
  @ColumnInfo(name = "name") val name: String,
  @ColumnInfo(name = "current") val current: Boolean,
  @ColumnInfo(name = "created_at") val createdAt: OffsetDateTime? = OffsetDateTime.now(ZoneId.of("UTC")),
  @ColumnInfo(name = "updated_at") val updatedAt: OffsetDateTime? = null
)

File: planik-flutter/planik/lib/local/entity/plan_entity.dart
import 'package:floor/floor.dart';
import 'package:flutter/material.dart';
import 'package:planik/local/entity/user_entity.dart';

@Entity(
  tableName: 'plans',
  foreignKeys: [
    ForeignKey(
      childColumns: ['user_id'],
      parentColumns: ['id'],
      entity: UserEntity,
      onDelete: ForeignKeyAction.cascade,
    )
  ],
  indices: [
    Index(value: ['user_id'])
  ],
)
@immutable
class PlanEntity {
  const PlanEntity({
    required this.userId,
    required this.name,
    required this.current,
    required this.createdAt,
    required this.updatedAt,
    this.id,
  });

  @PrimaryKey(autoGenerate: true)
  final int? id;
  @ColumnInfo(name: 'user_id')
  final int userId;
  final String name;
  final bool current;
  @ColumnInfo(name: 'created_at')
  final String createdAt;
  @ColumnInfo(name: 'updated_at')
  final String? updatedAt;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is PlanEntity &&
          runtimeType == other.runtimeType &&
          id == other.id &&
          userId == other.userId &&
          name == other.name &&
          current == other.current &&
          createdAt == other.createdAt &&
          updatedAt == other.updatedAt;

  @override
  int get hashCode =>
      id.hashCode ^
      userId.hashCode ^
      name.hashCode ^
      current.hashCode ^
      createdAt.hashCode ^
      updatedAt.hashCode;

  PlanEntity copyWith({
    int? id,
    int? userId,
    String? name,
    bool? current,
    String? createdAt,
    String? updatedAt,
    bool? archived,
  }) {
    return PlanEntity(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      name: name ?? this.name,
      current: current ?? this.current,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
